### PR TITLE
Make wolfSSL_CTX_set_timeout reflect to Session-ticket-lifetime-hint

### PR DESCRIPTION
**Pull request by TakayukiMatsuo**
If users want to change "session ticket lifetime hint", wolfSSL_CTX_set_TicketHint should be called. Since this API is wolfSSL specific, it seems to be difficult for openssl users to notice to use this API. This PR addresses followings:

- timeout value reflects to session-ticket-lifetime-hint. This behavior is same as OpenSSL
- wolfSSL_CTX_set_timeout returns different value from OpenSSL.
  If WOLFSSL_ERROR_CODE_OPENSSL defined, this API returns timeout value previouslly set.

### CODE REVIEW

**Review by cconlon — CHANGES_REQUESTED** [review]

File: src/ssl.c [hunk 1]
```diff
@@ -11057,7 +11057,14 @@ WOLFSSL_SESSION* wolfSSL_get_session(WOLFSSL* ssl)
     return NULL;
 }
 
-
+/**
+ * Performs a timeout check for the specified session based on the timeout
```
[the remark this answers is not in the document]
Comment (TakayukiMatsuo) [reply src/ssl.c:11061]: Added general functionality of this API to the function header.

File: src/ssl.c [hunk 2]
```diff
@@ -11057,7 +11057,14 @@ WOLFSSL_SESSION* wolfSSL_get_session(WOLFSSL* ssl)
...
+ * value specified by a server. Note that OpenSSL does not have such
+ * validation checks, so if an expired session is passed to this function,
+ * there will be a difference in the return value between wolfSSL and OpenSSL.
```
[the remark this answers is not in the document]
Comment (TakayukiMatsuo) [reply src/ssl.c:11064]: Added information about return value when the macro is defined.

File: src/ssl.c [hunk 3]
```diff
@@ -13382,18 +13389,65 @@ int wolfSSL_set_timeout(WOLFSSL* ssl, unsigned int to)
...
+ * Sets ctx session timeout in seconds.
+ * The timeout value set here should be reflected in the
+ * "session ticket lifetime hint" if this API works in the openssl compat-layer.
+ * Therefore wolfSSL_CTX_set_TicketHint is called internally.
+ * Arguments:
+ *  - ctx  WOLFSSL_CTX object which the timeout is set to
+ *  - to   timeout value in second
+ * Returns:
+ *  WOLFSSL_SUCCESS on success, BAD_FUNC_ARG on failure.
+ *  When WOLFSSL_ERROR_CODE_OPENSSL is defined, returns previous timeout value
```
[the remark this answers is not in the document]
Comment (TakayukiMatsuo) [reply src/ssl.c:13402]: Added information about return value when the macro is defined.

File: src/ssl.c [hunk 4]
```diff
@@ -11057,7 +11057,24 @@ WOLFSSL_SESSION* wolfSSL_get_session(WOLFSSL* ssl)
...
 
-
+/*
+ * Sets the session object to use when establishing a TLS/SSL session using
+ * the ssl object. Therefore, this function must be called before
+ * wolfSSL_connect. The session object to use can be obtained in a previous
+ * TLS/SSL connection using wolfSSL_get_session.
+ *
+ * This function rejects the session if it has been expired when this function
+ * is called. Note That this expiration check is wolfSSL specific and it brigns
```
Comment (cconlon) [inline src/ssl.c:11067]: Spelling - "brigns" should be "brings", but let's re-word that phrase to say "Note that this expiration check is wolfSSL specific and differs from OpenSSL return code behavior".

File: src/ssl.c [hunk 5]
```diff
@@ -11057,7 +11057,24 @@ WOLFSSL_SESSION* wolfSSL_get_session(WOLFSSL* ssl)
...
+ * return code difference from OpenSSL.
+ *
+ * By default, wolfSSL_set_session returns WOLFSSL_SUCCESS on successfully
+ * setting the session, WOLFSSL_FAILURE on failure due to the session cache
+ * being disabled, or the session has expired.
+ *
+ * If you want to get the same return code when the session has expired,
```
Comment (cconlon) [inline src/ssl.c:11074]: Let's reword this as "To match OpenSSL return code behavior when session is expired, define OPENSSL_EXTRA and WOLFSSL_ERROR_CODE_OPENSSL. This behavior will return WOLFSSL_SUCCESS even when the session is expired and rejected."

**Review by cconlon — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 4 above)
Comment (TakayukiMatsuo) [reply src/ssl.c:11067]: Replaced the statements.

File: src/ssl.c (hunk 5 above)
Comment (TakayukiMatsuo) [reply src/ssl.c:11074]: Replaced the statements.

**Comment by TakayukiMatsuo** [comment]
Finished cleaning up the commit history.

**Comment by JacobBarthelmeh** [comment]
Thanks @TakayukiMatsuo , am investigating into the change of return value behavior with an ABI tagged function. I know the change here is limited to a very specific build but am determining if we may have to have a different compatibility layer function that wraps around wolfSSL_CTX_set_timeout() so that in all build cases its return remains constant.

**Review by JacobBarthelmeh — APPROVED** [review]

**Review by cconlon — APPROVED** [review]
